adal/src/main/java/com/microsoft/aad/adal/BasicWebViewClient.java

```diff
@@ -150,11 +150,37 @@ public void onPageFinished(WebView view, String url) {
 
     @Override
     public void onPageStarted(WebView view, String url, Bitmap favicon) {
-        Logger.v(TAG + "onPageStarted", "page is started with the url " + url);
+        logPageStartLoadingUrl(url);
         super.onPageStarted(view, url, favicon);
         showSpinner(true);
     }
 
+    private void logPageStartLoadingUrl(final String url) {
+        if (TextUtils.isEmpty(url)) {
+            Logger.v(TAG, "onPageStarted: Null url for page to load.");
+            return;
+        }
+
+        final Uri uri = Uri.parse(url);
+        if (uri.isOpaque()) {
+            Logger.v(TAG, "onPageStarted: Non-hierarchical loading uri: " + url);
+            return;
+        }
+
+        if (doesLoadingUrlContainAuthCode(uri)) {
+            Logger.v(TAG, "Webview starts loading: " + uri.getHost() + uri.getPath()
+                    + " Auth code is returned for the loading url.");
+        } else {
+            Logger.v(TAG, "Webview starts loading: " + uri.getHost() + uri.getPath(),
+                    "Full loading url is: " + url, null);
+        }
+    }
+    
+    private boolean doesLoadingUrlContainAuthCode(final Uri uri) {
```

Comment:
I'm not following. I want to log the redirect which is helpful when debugging, but i cannot log auth code.